samples/nrf9160/modem_shell/src/cloud/cloud_shell.c

```diff
@@ -67,6 +75,63 @@ static void notify_pgps(struct k_work *work)
 }
 #endif /* defined(CONFIG_NRF_CLOUD_PGPS) */
 
+static void cloud_cmd_execute(struct k_work *work)
+{
+	const struct shell *shell = shell_backend_uart_get_ptr();
+
+	shell_execute_cmd(shell, shell_cmd);
+	memset(shell_cmd, 0, CLOUD_CMD_MAX_LENGTH);
+}
+
+static bool cloud_shell_parse_mosh_cmd(const char *buf_in)
+{
+	const cJSON *app_id = NULL;
+	const cJSON *mosh_cmd = NULL;
+	bool ret = false;
+
+	cJSON *cloud_cmd_json = cJSON_Parse(buf_in);
+
+	/* A format example expected from nrf cloud:
+	 * {"appId":"MODEM_SHELL", "data":"location get"}
+	 */
+	if (cloud_cmd_json == NULL) {
+		const char *error_ptr = cJSON_GetErrorPtr();
+
+		if (error_ptr != NULL) {
+			mosh_error("JSON parsing error before: %s\n", error_ptr);
+		}
+		ret = false;
+		goto end;
+	}
+
+	/* MoSh commands are identified by checking if appId equals "MODEM_SHELL" */
+	app_id = cJSON_GetObjectItemCaseSensitive(cloud_cmd_json, "appId");
+	if (cJSON_IsString(app_id) && (app_id->valuestring != NULL)) {
+		if (strcmp(app_id->valuestring, "MODEM_SHELL") != 0) {
+			ret = false;
+			goto end;
+		}
+	}
+
+	/* TODO: Consider adding similar block for checking that messageType == "CMD" */
```

Comment:
Is this TODO still valid? Shall we take messageType in use?

Reply:
Taking into account jayteemo's previous comment, I don't think so. No need to go crazy with the json attributes. I'll remove it.